### PR TITLE
Ignore invalid arguments to avoid 500 in API

### PR DESCRIPTION
**Pull request by theefer**
Next step will be to 400 with an error when they are invalid.

### CODE REVIEW

**Comment by kenoir** [comment]
:+1:

File: media-api/app/controllers/MediaApi.scala [hunk 1]
```diff
@@ -341,25 +345,25 @@ object SearchParams {
...
       request.getQueryString("missingIdentifier"),
-      request.getQueryString("valid").map(_.toBoolean),
-      request.getQueryString("free").map(_.toBoolean),
+      request.getQueryString("valid") flatMap parseBooleanFromQuery,
+      request.getQueryString("free") flatMap parseBooleanFromQuery,
       request.getQueryString("uploadedBy"),
       commaSep("labels"),
       commaSep("hasMetadata"),
-      request.getQueryString("costModelDiff").nonEmpty
+      request.getQueryString("costModelDiff") flatMap (s => Try(s.toBoolean).toOption) getOrElse false
```
Comment (jamesgorrie) [inline media-api/app/controllers/MediaApi.scala]: Assuming this is now a `Boolean` in the QS e.g. `?costModelDiff=true`?

Comment (jamesgorrie) [inline media-api/app/controllers/MediaApi.scala]: If so - why don't we use the `parseBooleanFromQuery`?

Comment (theefer) [inline media-api/app/controllers/MediaApi.scala]: Thanks, oversight, will add!

Comment (jamesgorrie) [inline media-api/app/controllers/MediaApi.scala]: So it is now a bool val with default as false?

Comment (theefer) [inline media-api/app/controllers/MediaApi.scala]: Yes. That way our query params are uniform

**Pull request merged**